src/DataTypes/Serializations/ISerialization.cpp

```diff
@@ -197,9 +203,19 @@ String getNameForSubstreamPath(
         else if (it->type == Substream::VariantOffsets)
             stream_name += ".variant_offsets";
         else if (it->type == Substream::VariantElement)
-            stream_name += "." + it->variant_element_name;
+        {
+            if (escape_for_file_name && escape_variant_substreams)
```

Comment:
Sorry not clear why we check both flags here?

Reply:
If `escape_for_file_name = false` we don't need to escape anything. This function can be used not only for creating file names, so escaping variant substream with disabled `escape_for_file_name` doesn't make sense